arangod/Agency/Supervision.cpp

```diff
@@ -2219,6 +2225,88 @@ void Supervision::checkBrokenAnalyzers() {
   }
 }
 
+
+void Supervision::checkReplicatedLogs() {
+  _lock.assertLockedByCurrentThread();
+
+  using namespace replication2::agency;
+
+  auto const readPlanSpecification = [](Node const& node) -> LogPlanSpecification {
+    auto builder = node.toBuilder();
+    return LogPlanSpecification{from_velocypack, builder.slice()};
+  };
+
+  auto const readLogCurrent = [](Node const& node) -> LogCurrent {
+    auto builder = node.toBuilder();
+    return LogCurrent(from_velocypack, builder.slice());
+  };
+
+  // check if Plan has replicated logs
+  auto const& [node, exists] = snapshot().hasAsNode(planRepLogPrefix);
+  if (!exists) {
+    return;
+  }
+
+  using namespace replication2::algorithms;
+
+  ParticipantInfo info = std::invoke([&]{
+    ParticipantInfo info;
+    Node::Children const& dbservers = snapshot().hasAsChildren(plannedServers).first;
+    for (auto const& [serverId, node]: dbservers) {
+      bool const isHealthy = serverHealth(serverId) == HEALTH_STATUS_GOOD;
+      std::pair<uint64_t, bool> rebootID =
+          snapshot().hasAsUInt(curServersKnown + serverId + "/" + StaticStrings::RebootId);
+      if (rebootID.second) {
+        info.emplace(serverId, ParticipantRecord{RebootId{rebootID.first}, isHealthy});
+      }
+    }
+    return info;
+  });
+
+  auto builder = std::make_shared<Builder>();
```

Comment:
We could also turn this into `Builder builder`, so we won't have an initial memory allocation for the shared_ptr.

Reply:
Sadly we can't, because `_agent->write` expects a `query_t` which is a typedef for `std::shared_ptr<arangodb::velocypack::Builder>` 😞